go/libkb/globals.go

```diff
@@ -226,7 +228,7 @@ func (g *GlobalContext) createLoginStateLocked() {
 		g.loginState.Shutdown()
 	}
 	g.loginState = NewLoginState(g)
-	g.ActiveDevice = new(ActiveDevice)
+	g.ActiveDevice.clear(g.loginState.account)
```

Comment:
I was thinking that since g.loginState is new, there will be no contention.

But I'm fine with your reassigning fix as well.